delfin/alert_manager/snmp_validator.py

```diff
@@ -45,12 +45,18 @@ def validate(self, ctxt, alert_source):
             # engine id if engine id is empty. Therefore, engine id
             # should be saved in database.
             if not engine_id and alert_source.get('engine_id'):
-                db.access_info_update(ctxt,
-                                      alert_source.get('storage_id'),
-                                      alert_source)
 
+                alert_source_dict = {
+                    'engine_id': alert_source.get('engine_id')}
+                db.alert_source_update(ctxt,
+                                       alert_source.get('storage_id'),
+                                       alert_source_dict)
+            self._handle_validation_result(ctxt,
```

Comment:
Category.RECOVERY shoul be send only if previos attempt was FAULT , right? We are sending a Category.RECOVERY , even for the first attempt !!

Reply:
yes, do you have any good suggestion on it? such as remember the fault event?